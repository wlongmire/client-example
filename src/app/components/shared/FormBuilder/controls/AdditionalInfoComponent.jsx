import React, { PropTypes } from 'react'

export function AdditionalInfoComponent(props) {
  const addInfoColor = props.additionalInfo1Color ? props.additionalInfo1Color : '#417505'

  if (props.additionalInfo1 && props.additionalInfo2 && props.additionalInfoIcon) {
    return (
      <div
        className="additional-info-div"
      >
        <div
          className="additional-info-row-1"
          style={{ color: addInfoColor }}
        ><i className="material-icons">{props.additionalInfoIcon}</i>
          <span className="additional-info-1"><b>{props.additionalInfo1}</b></span>
        </div>
        <br />
        <div className="additional-info-row-2">{props.additionalInfo2}</div>
      </div>)
  }

  return (<div />)
}

export default AdditionalInfoComponent

AdditionalInfoComponent.propTypes = {
  additionalInfo1Color: PropTypes.string,
  additionalInfo1: PropTypes.string,
  additionalInfo2: PropTypes.string,
  additionalInfoIcon: PropTypes.string
}