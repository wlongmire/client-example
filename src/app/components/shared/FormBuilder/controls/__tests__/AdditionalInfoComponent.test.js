/* eslint-env node, jest */

import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { AdditionalInfoComponent } from '../AdditionalInfoComponent'

describe('>>> AdditionalInfoComponent Component --- Snapshot', () => {
  it('+++ provide a dropdown field', () => {
    const renderedValue = shallow(
      <AdditionalInfoComponent
        additionalInfoIcon={'Additional Info Icon'}
        additionalInfo1={'row 111'}
        additionalInfo2={'row 21123123'}
        additionalInfo1Color={'blue'}
      />)
    expect(toJson(renderedValue)).toMatchSnapshot()
  })
})