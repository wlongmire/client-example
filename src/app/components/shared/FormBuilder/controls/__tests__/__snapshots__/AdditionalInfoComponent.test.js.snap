// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> AdditionalInfoComponent Component --- Snapshot +++ provide a dropdown field 1`] = `
<div
  className="additional-info-div"
>
  <div
    className="additional-info-row-1"
    style={
      Object {
        "color": "blue",
      }
    }
  >
    <i
      className="material-icons"
    >
      Additional Info Icon
    </i>
    <span
      className="additional-info-1"
    >
      <b>
        row 111
      </b>
    </span>
  </div>
  <br />
  <div
    className="additional-info-row-2"
  >
    row 21123123
  </div>
</div>
`;
